docs/community/20241112-node-bootstrapping.md

```diff
@@ -0,0 +1,121 @@
+---
+title: ClusterAPI Node Bootstrapping Working Group
+authors:
+    - "@t-lo"
+reviewers:
+    - "@elmiko"
+    - "@eljohnson92"
+    - "@johananl"
+    - "@tormath1"
+    - "@fabriziopandini"
+    - "@sbueringer"
+
+creation-date: 2024-11-12
+last-updated: 2024-11-15
+status: proposed
+see-also:
+  - https://github.com/kubernetes-sigs/cluster-api/issues/5294
+  - https://docs.google.com/document/d/1Fz5vWwhWA-d25_QDqep0LWF6ae0DnTqd5-8k8N0vDDM/edit
+  - https://github.com/kubernetes-sigs/cluster-api/issues/9157
+  - https://youtu.be/0AhA4Box3MM?si=IfKJfMWmlA9EW7ri&t=172
+  - https://docs.google.com/document/d/12v6NFr7xal9RH3GEB_1IR6HENq2GMuzALS3l5elvrMg/edit
+---
+
+# Node Bootstrapping Working Group ("CAPI-NoBo")
+
+This document briefly outlines the scope, communication media, and
+stakeholders for a Working Group dedicated to the Provisioning aspect of
+Node Bootstrapping.
+
+Provisioning, in this context, refers to configuring and customising the
+node operating system to prepare it to serve as a ClusterAPI cluster node.
+
+
+## User Story and Problem Statement
+
+As a Linux Distribution maintainer aiming to support ClusterAPI, or to improve
+my existing support of ClusterAPI in my distribution, I would like to be able to
+cleanly integrate my contributions into the node bootstrapping process without
+interfering with other implementations.
+I would like my contributions to be as re-usable as possible across different
+bootstrap providers.
+I would like to read and to follow documentation, guidelines, and specifications
+on the above.
+I would like to offer a choice of node bootstrapping configuration systems to users,
```

Comment:
can we include an illustrative example? e.g "I would like to enable the bootstrap API consumer to express intent to fine tune nodes by setting vm.dirty_ratio while not having to reimplement the logic to setup kubelet and let the machine become a Node" if that's a good representative?

Reply:
This is a great idea!
Probably the not a perfect example though, as setting sysctls solely involves the OS layer - i.e. it does not strictly fulfil a provisioning need of the bootstrap provider. IOW we can provision the nodes perfectly fine and alternatively set the sysctl later, e.g via an operator. Also, this opens a whole can of works as we would later need to validate the intent (and let the provisioning fail if it doesn't work) etc. 

How about something more fundamentally provisioning related like disks set-up or disk partitioning? E.g. adding a required storage device / file system to a mount point? I'll add something in the doc.